File: tests_bench/objects.lua
local start = os.clock()

local cat = {
    name = "meow",
    age = 5,
    color = "black"
}

local sum = 0
local N = 100000000


for i = 1, N do
    sum = sum + cat.age
end

local elapsed = os.clock() - start

print("sum:", sum)
print("time:", elapsed)

File: tests_bench/objects.py
import time

start = time.perf_counter()


cat = {"name": "Whiskers", "age": 5, "color": "black"}


sum_ = 0
N = 100_000_000

for _ in range(N):
    sum_ += cat["age"]

end = time.perf_counter()

print("sum:", sum_)
print("time:", end - start)

